### ISSUE TITLE
Confused by error: calling 'unison()' at this point skips initialization of a needed variable

### ISSUE THREAD

**Issue by heuermh** [ccrma/chuck#348]
I'm not able to figure out what the compiler is trying to tell me here, it used to work previous to ChucK version 1.5.1.0
```bash
$ chuck --loop &
$ chuck + import.ck
...
Chord.ck:30:15: error: calling 'unison()' at this point skips initialization of a needed variable:
[30]     Intervals.unison() @=> Interval unison;
                   ^
Chord.ck:30:31: error: ...(note: this skipped variable initialization is needed by 'fun Unison Intervals.unison()')
[30]     Intervals.unison() @=> Interval unison;
                                   ^
Chord.ck:47:11907: error: ...(note: this is where the variable is used within 'fun Unison Intervals.unison()' or its subsidiaries)
Chord.ck: ...(hint: try calling 'unison()' after the variable initialization)
```

Chord.ck https://github.com/heuermh/lick/blob/master/lick/interval/Chord.ck#L23
Intervals.ck https://github.com/heuermh/lick/blob/master/lick/interval/Intervals.ck#L26

I tried to create a small test case, but this works
```bash
$ cat Baz.ck
class Qux
{
  fun static Qux create()
  {
    Qux qux;
    return qux;
  }
}
class Baz
{
  Qux.create() @=> Qux qux;

  fun Qux sample()
  {
    return qux;
  }
}

$ chuck Baz.ck
```

**Comment by gewang** [comment]
This bug appears to be caused by the compiler not taking into account values/func/types created in a different file context and treating them as if they were part of the current file context. A fix has been pushed in fc1d609

**Comment by heuermh** [comment]
Thanks!  Will give it a shot

**Comment by gewang** [comment]
FYI encountering a crash on windows related to the newly added code, investigating...

**Comment by gewang** [comment]
crash remedy pushed ccb5af8; let us know if you run into any further funny business hah

**Comment by heuermh** [comment]
After a few more fixes to LicK, I've only https://github.com/ccrma/chugins/issues/70 left, and that is already being looked into.  Thanks again!

**Comment by gewang** [comment]
> Progress made, thank you!
> 
> Now on mac at commit [2295db4](https://github.com/ccrma/chuck/commit/2295db40f89b69a3c924022ce081d55181e0031f)
> 
> ```
> [chuck]: (VM) sporking incoming shred: 188 (IntervalDurProvider.ck)...
> Pattern.ck:33:9: error: not all control paths in 'fun int Pattern.size(int capacity)' return a value
> [33]     fun int size(int capacity)
>              ^
> Segmentation fault: 11
> ```
> 
> Will continue to work through these . . .

Thanks Michael! Hmm, that segmentation fault is an issue; if you'd like, could you reproduce and send code? We should try to fix that!

**Comment by gewang** [comment]
update: I am able to repro the crash from LiCK `main` branch and running `import.ck`... enough to start investigating; will report back

**Comment by gewang** [comment]
located source of the segmentation fault; fix(hopefully!) pushed in bd8b168

**Comment by heuermh** [comment]
Yes, after that commit, I confirm no more seg faults

**Comment by gewang** [comment]
FYI ran latest in tip of `main` of LiCK with success (macOS Ventura 13.5)
```
chuck import.ck test.ck
[...]
"ArrayListTest ok" :(string)
"ArraySetTest ok" :(string)
"GraphTest ok" :(string)
"GraphsTest ok" :(string)
"FloatArrayListTest ok" :(string)
"IntArrayListTest ok" :(string)
"InterpolationTest ok" :(string)
"LoopsTest ok" :(string)
```

**Comment by heuermh** [comment]
Thank you, @gewang!  Closing, fixed as of 1.5.1.1 release
